client/src/utils.rs

```diff
@@ -100,19 +106,20 @@ pub fn contains_sudo_pallet(metadata: &Metadata) -> bool {
 }
 
 /// Checks if the account has sufficient funds. Exits the process if not.
-pub fn ensure_payment(api: &Api, xt: &str, tx_payment_cid: Option<&str>) {
+pub fn ensure_payment(api: &Api, encoded_xt: &str, tx_payment_cid: Option<&str>) {
 	if let Some(cid_str) = tx_payment_cid {
-		ensure_payment_cc(api, cid_str, xt);
+		ensure_payment_cc(api, cid_str, encoded_xt);
 	} else {
-		ensure_payment_native(api, xt);
+		ensure_payment_native(api, encoded_xt);
 	}
 }
 
-fn ensure_payment_cc(api: &Api, cid_str: &str, xt: &str) {
+fn ensure_payment_cc(api: &Api, cid_str: &str, encoded_xt: &str) {
```

Comment:
Can't we pass a borrow of the unencoded xt?

Reply:
I wanted to have the borrowing fixed after the CI passed, but unfortunately I had not gotten to that part (yet).